File: src/main/java/org/dedda/bratwurst/lang/VariableDeclaration.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;

/**
 * Created by dedda on 10/15/15.
 *
 * @author dedda
 */
public class VariableDeclaration extends BWInstruction {

    private final String variableName;
    private final BWExpression targetValue;

    public VariableDeclaration(int lineNumber, String variableName, BWExpression targetValue) {
        super(lineNumber);
        this.variableName = variableName;
        this.targetValue = targetValue;
    }

    @Override
    public void run(Scope scope) {
        targetValue.run(scope);
        BWObject value = targetValue.getValue();
        BWVariable variable = new BWVariable(variableName, value);
        scope.setVariable(variable);
    }

    public String getVariableName() {
        return variableName;
    }
}

File: src/main/java/org/dedda/bratwurst/lang/VariableDeclaration.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 10/15/15.
 *
 * @author dedda
 */
class VariableDeclaration(lineNumber: Int, val variableName: String, private val targetValue: BWExpression) : BWInstruction(lineNumber) {
    override fun run(scope: Scope) {
        targetValue.run(scope)
        val value = targetValue.value
        val variable = BWVariable(variableName, value)
        scope.setVariable(variable)
    }

}